src/inference/src/ie_core.cpp

```diff
@@ -540,6 +541,7 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             bool loadedFromCache = false;
             auto hash = CalculateFileHash(modelPath, parsed._deviceName, plugin, parsed._config);
             auto lock = cacheGuard.getHashLock(hash);
+            parsed._config[InferenceEngine::PluginConfigParams::KEY_CACHE_DIR] = coreConfig.getCacheConfig()._cacheDir;
```

Comment:
CC @nosovmik

Reply:
why do we need it?

Reply:
Not sure that this will work for other plugins (like GNA, KMB or Myriad) - they may throw exception if such key is not supported.

Reply:
@ilya-lavrenov @nosovmik, thanks a lot for pointing this out! CACHE_DIR is indeed passed to plugins in [GetCPPPluginByName](https://github.com/openvinotoolkit/openvino/blob/e64f94ed7ecb07a7635ec4ccccbf5e4be75e3736/src/inference/src/ie_core.cpp#L777-L782), so I reverted changes to ie_core and updated CPU config. More details in "[§] Revert changes to ie_core. CACHE_DIR is always in SUPPORTED_CONFIG_KEYS" ([link](https://github.com/openvinotoolkit/openvino/pull/7936/commits/e64f94ed7ecb07a7635ec4ccccbf5e4be75e3736)).